fpmsyncd/routesync.cpp

```diff
@@ -109,24 +109,29 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
         /* 0 means no ifindex */
         if (m_link_cache)
         {
-            rtnl_link_i2name(m_link_cache, ifindex, addrStr, MAX_ADDR_SIZE);
-            ifindexes += addrStr;
+            rtnl_link_i2name(m_link_cache, ifindex, ifname, MAX_ADDR_SIZE);
+            if (!strlen(ifname))
+            {
+                rtnl_link_alloc_cache(m_nl_sock, AF_UNSPEC, &m_link_cache);
+                rtnl_link_i2name(m_link_cache, ifindex, ifname, MAX_ADDR_SIZE);
```

Comment:
could the ifname still not get resolved even after updating the cache? 
should we throw exception or print here if it happens?